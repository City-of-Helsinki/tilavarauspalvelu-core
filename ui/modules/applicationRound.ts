import { i18n } from "next-i18next";
import { ApplicationRoundType } from "./gql-types";
import { capitalize } from "./util";

export const getApplicationRoundName = (
  applicationRound: ApplicationRoundType,
  language: string = i18n.language
): string => {
  const key = `name${capitalize(language)}`;
  return applicationRound[key] || applicationRound.nameFi;
};
