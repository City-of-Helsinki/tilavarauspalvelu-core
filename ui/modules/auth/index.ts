export { default as signOut } from "./signout";
