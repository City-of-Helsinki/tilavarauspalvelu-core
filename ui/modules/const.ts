import { OptionType } from "./types";

export const weekdays = [
  "monday",
  "tuesday",
  "wednesday",
  "thursday",
  "friday",
  "saturday",
  "sunday",
];

export const reservationUnitPrefix = "/reservation-unit";
export const searchPrefix = "/search";
export const applicationsPrefix = "/applications";
export const applicationPrefix = "/application";
export const criteriaPrefix = "/criteria";

export const mapStyle = {
  version: 8,
  name: "hel-osm-light",
  metadata: {},
  sources: {
    osm: {
      type: "raster",
      tiles: ["https://tiles.hel.ninja/styles/hel-osm-light/{z}/{x}/{y}.png"],
      minzoom: 0,
      maxzoom: 20,
    },
  },
  layers: [
    {
      id: "osm",
      type: "raster",
      source: "osm",
    },
  ],
  id: "hel-osm-light",
};

export const reservationUnitPath = (id: number): string =>
  `${reservationUnitPrefix}/${id}`;

export const emptyOption = (label: string): OptionType => ({
  label,
  value: undefined,
});

export const participantCountOptions = [
  1,
  2,
  5,
  10,
  15,
  20,
  25,
  30,
  40,
  50,
  60,
  80,
  100,
  150,
  200,
].map((v) => ({ label: `${v}`, value: v } as OptionType));

const option = (label: string, value: string): OptionType => {
  return { label, value };
};

const formatNumber = (n: number): string => `00${n}`.slice(-2);

export const durationOptions = [] as OptionType[];

let h = 1;
let m = 30;

for (let i = 0; i < 12; i += 1) {
  durationOptions.push(
    option(
      `${h} ${h === 1 ? "tunti" : "tuntia"} ${m ? `${m} min` : ""}`,
      `${formatNumber(h)}:${formatNumber(m)}:00`
    )
  );
  m += 15;
  if (m === 60) {
    m = 0;
    h += 1;
  }
}
export const defaultDuration = "01:30:00";

export const isBrowser = typeof window !== "undefined";

export const sentryDSN = process.env.NEXT_PUBLIC_SENTRY_DSN;

export const matomoEnabled = process.env.NEXT_PUBLIC_ENABLE_MATOMO === 'true';

export const sentryEnvironment = process.env.NEXT_PUBLIC_SENTRY_ENVIRONMENT;

export const apiBaseUrl = isBrowser ? window.config?.apiBaseUrl : process.env.TILAVARAUS_API_URL;

export const authEnabled = isBrowser ? window.config?.authEnabled: process.env.DISABLE_AUTH !== "true";

export const oidcClientId = process.env.NEXT_PUBLIC_OIDC_CLIENT_ID;

export const oidcUrl = process.env.NEXT_PUBLIC_OIDC_URL;

export const oidcScope = process.env.NEXT_PUBLIC_OIDC_SCOPE;

export const apiScope = process.env.NEXT_PUBLIC_TILAVARAUS_API_SCOPE;
