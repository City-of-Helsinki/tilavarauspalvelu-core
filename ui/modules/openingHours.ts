import { isWithinInterval } from "date-fns";
import { uniq } from "lodash";
import { i18n } from "next-i18next";
import { PeriodType, TimeSpanType } from "./gql-types";
import { toApiDate } from "./util";

export type ActiveOpeningTime = {
  day: number;
  label: string;
  from: string;
  to: string;
};

export type OpeningHourRow = {
  label: string;
  value: string | number;
  index: number;
};

const reservableStates = ["open"];

export const getActiveOpeningTimePeriod = (
  openingTimePeriods: PeriodType[],
  date: string
): PeriodType | undefined => {
  return openingTimePeriods?.find((openingTimePeriod) => {
    const startDate = new Date(openingTimePeriod.startDate);
    const endDate = new Date(openingTimePeriod.endDate);
    return (
      isWithinInterval(new Date(date), {
        start: startDate,
        end: endDate,
      }) ||
      ((openingTimePeriod.startDate === null || toApiDate(startDate) <= date) &&
        (openingTimePeriod.endDate === null || toApiDate(endDate) >= date))
    );
  });
};

export const getActiveOpeningTimes = (
  openingTimePeriods?: PeriodType[]
): ActiveOpeningTime[] => {
  const result = [] as ActiveOpeningTime[];
  const activeOpeningTimePeriod = getActiveOpeningTimePeriod(
    openingTimePeriods,
    toApiDate(new Date())
  );

  const timeSpans = activeOpeningTimePeriod?.timeSpans?.filter((timeSpan) =>
    reservableStates.includes(timeSpan.resourceState)
  );
  const weekdays = uniq(
    timeSpans?.reduce((acc, timeSpan) => acc.concat(timeSpan.weekdays), [])
  ).sort();
  weekdays.forEach((weekday) => {
    const activeTimeSpans: TimeSpanType[] = timeSpans?.filter((n) =>
      n.weekdays.includes(weekday)
    );
    activeTimeSpans.forEach((timeSpan) => {
      result.push({
        day: weekday,
        label: i18n.t(`common:weekDay.${weekday}`),
        from: timeSpan.startTime,
        to: timeSpan.endTime,
      });
    });
  });

  return result;
};

export const getDayOpeningTimes = (
  openingTime: { label: string; from: string; to: string },
  index: number
): OpeningHourRow => {
  const { label, from, to } = openingTime;
  const [fromHours, fromMinutes] = from.split(":").map(Number);
  const [toHours, toMinutes] = to.split(":").map(Number);

  const fromDate = new Date();
  fromDate.setUTCHours(fromHours, fromMinutes);
  const fromStr = i18n.t("common:time", { date: fromDate });

  const toDate = new Date();
  toDate.setUTCHours(toHours, toMinutes);
  const toStr = i18n.t("common:time", { date: toDate });

  return { label, value: `${fromStr} - ${toStr}`, index };
};
