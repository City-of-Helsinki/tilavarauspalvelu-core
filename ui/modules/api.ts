import { AxiosRequestConfig, AxiosResponse } from "axios";
import { apiBaseUrl } from "./const";
import axiosClient from "./auth/axiosClient";
import {
  Application,
  ApplicationRound,
  ReservationUnit,
  Parameter,
  Reservation,
  RecurringReservation,
  User,
  ApplicationStatusChange,
} from "./types";
import { ApiError } from "./ApiError";

const applicationRoundBasePath = "application_round";
const reservationUnitsBasePath = "reservation_unit";
const reservationBasePath = "reservation";
const recurringReservationBasePath = "recurring_reservation";
const parameterBasePath = "parameters";
const applicationBasePath = "application";
const applicationEventStatusBasePath = "application_event_status";
const applicationEventFeedBasePath = "application_event_calendar";
const userBasePath = "users";

// eslint-disable-next-line @typescript-eslint/no-empty-interface
interface QueryParameters {}

interface RequestParameters {
  path: string;
  headers?: { [key: string]: string };
  parameters?: QueryParameters;
  data?: Application;
}

enum ApiResponseFormat {
  json = "json",
}

interface ApiParameters extends QueryParameters {
  format: ApiResponseFormat;
}

async function request<T>(requestConfig: AxiosRequestConfig): Promise<T> {
  const config: AxiosRequestConfig = requestConfig;

  try {
    const response: AxiosResponse<T> = await axiosClient.request<
      T,
      AxiosResponse<T>
    >(config);
    return response.data;
  } catch (error) {
    const errorMessage: string | undefined = error.response?.data?.detail;
    if (errorMessage) {
      throw new ApiError(errorMessage, error.response?.status);
    } else {
      throw new ApiError("ApiError 'no message'", error.response?.status);
    }
  }
}

async function apiGet<T>({
  path,
  parameters = {} as QueryParameters,
}: RequestParameters): Promise<T> {
  const apiParameters: ApiParameters = {
    ...parameters,
    format: ApiResponseFormat.json,
  };

  return request<T>({
    url: `${apiBaseUrl}/${path}/`,
    headers: {
      "Content-Type": "application/json",
    },
    method: "get",
    params: apiParameters,
  });
}

const validateStatus = (status: number): boolean => status < 300;

async function apiPut<T>({ path, data }: RequestParameters): Promise<T> {
  return request<T>({
    url: `${apiBaseUrl}/${path}`,
    headers: {
      "Content-Type": "application/json",
    },
    method: "put",
    data,
    validateStatus,
  });
}

async function apiPost<T>({ path, data }: RequestParameters): Promise<T> {
  return request<T>({
    url: `${apiBaseUrl}/${path}`,
    headers: {
      "Content-Type": "application/json",
    },
    method: "post",
    data,
    validateStatus,
  });
}

export function getApplicationRounds(): Promise<ApplicationRound[]> {
  return apiGet<ApplicationRound[]>({
    path: `v1/${applicationRoundBasePath}`,
  });
}

export function getApplicationRound(
  params: IDParameter
): Promise<ApplicationRound> {
  return apiGet<ApplicationRound>({
    path: `v1/${applicationRoundBasePath}/${params.id}`,
  });
}

export interface ReservationUnitsParameters {
  applicationRound?: number;
  application?: number;
  search?: string;
  purpose?: number;
  reservationUnitType?: number;
  unit?: number;
  limit?: number;
  after?: string;
}

export function getReservationUnits(
  params: QueryParameters
): Promise<ReservationUnit[]> {
  return apiGet<ReservationUnit[]>({
    parameters: params,
    path: `v1/${reservationUnitsBasePath}`,
  });
}

export function getReservations(): Promise<Reservation[]> {
  return apiGet<Reservation[]>({
    path: `v1/${reservationBasePath}`,
  });
}

export function getRecurringReservations(
  applicationId: number
): Promise<RecurringReservation[]> {
  return apiGet<RecurringReservation[]>({
    path: `v1/${recurringReservationBasePath}`,
    parameters: { application: applicationId },
  });
}

export async function getDecisionMaker(
  applicationEventId: number
): Promise<User | null> {
  const statusChanges = await apiGet<ApplicationStatusChange[]>({
    path: `v1/${applicationEventStatusBasePath}`,
    parameters: { application: applicationEventId },
  });

  const decisionStatusChange = statusChanges
    .filter(
      (sc) =>
        sc.applicationId === applicationEventId && sc.status === "review_done"
    )
    .sort(
      (a, b) =>
        new Date(b.timestamp).getTime() - new Date(a.timestamp).getTime()
    )
    .find(() => true);

  return apiGet<User>({
    path: `v1/${userBasePath}/${decisionStatusChange?.userId}`,
  });
}

interface IDParameter {
  id: number;
}

export function getReservationUnit(id: number): Promise<ReservationUnit> {
  return apiGet<ReservationUnit>({
    path: `v1/${reservationUnitsBasePath}/${id}`,
  });
}

export function getParameters(
  name:
    | "purpose"
    | "age_group"
    | "ability_group"
    | "reservation_unit_type"
    | "district"
    | "city"
): Promise<Parameter[]> {
  return apiGet<Parameter[]>({
    path: `v1/${parameterBasePath}/${name}`,
  });
}

export function getApplication(id: number): Promise<Application> {
  return apiGet<Application>({
    path: `v1/${applicationBasePath}/${id}`,
  });
}

export function getApplications(): Promise<Application[]> {
  return apiGet<Application[]>({
    path: `v1/${applicationBasePath}`,
  });
}

export function saveApplication(
  application: Application
): Promise<Application> {
  if (application.id === undefined) {
    return apiPost<Application>({
      data: application,
      path: `v1/${applicationBasePath}/`,
    });
  }
  return apiPut<Application>({
    data: application,
    path: `v1/${applicationBasePath}/${application.id}/`,
  });
}

export const cancelApplication = async (
  applicationId: number
): Promise<void> => {
  const application = await getApplication(applicationId);
  application.status = "cancelled";
  await saveApplication(application);
};

export const applicationEventCalendarFeedUrl = (uuid: string): string =>
  `${apiBaseUrl}/v1/${applicationEventFeedBasePath}/${uuid}`;
