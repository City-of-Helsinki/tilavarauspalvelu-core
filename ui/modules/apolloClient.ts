import {
  ApolloClient,
  ApolloLink,
  fromPromise,
  HttpLink,
  InMemoryCache,
  from,
} from "@apollo/client";
import { relayStylePagination } from "@apollo/client/utilities";
import { onError } from "@apollo/client/link/error";
import {
  getAccessToken,
  getApiAccessToken,
  updateApiAccessToken,
} from "./auth/util";
import { apiBaseUrl } from "./const";

// list of operations that need authentication
const needsAuthentication = ["listReservations", "reservationByPk", "units"];

const getNewToken = (): Promise<string> =>
  updateApiAccessToken(getAccessToken());

const authLink = new ApolloLink((operation, forward) => {
  const token = getApiAccessToken();

  if (!token) {
    return fromPromise(
      getNewToken().catch(() => {
        // TODO Handle token refresh error
        if (needsAuthentication.includes(operation.operationName)) {
          return false;
        }

        return forward(operation);
      })
    )
      .filter((value) => Boolean(value))
      .flatMap((accessToken) => {
        const oldHeaders = operation.getContext().headers;
        operation.setContext({
          headers: {
            ...oldHeaders,
            authorization: `Bearer ${accessToken}`,
          },
        });

        return forward(operation);
      });
  }

  operation.setContext(({ headers }) => ({
    headers: {
      ...headers,
      authorization: token ? `Bearer ${token}` : "",
    },
  }));

  return forward(operation);
});

// eslint-disable-next-line consistent-return
const errorLink = onError(({ graphQLErrors, operation, forward }) => {
  if (graphQLErrors) {
    const authError = graphQLErrors.find((e) => {
      return (
        e.message !== null &&
        (e.message.indexOf("AnonymousUser") !== -1 ||
          e.message.indexOf("has expired") !== -1 ||
          e.message.indexOf("too old") !== -1 ||
          e.message.indexOf("No permission to mutate") !== -1)
      );
    });

    const hasAuthError = Boolean(authError !== undefined);

    if (hasAuthError) {
      return fromPromise(
        getNewToken().catch(() => {
          // TODO Handle token refresh error
          return null;
        })
      )
        .filter((value) => Boolean(value))
        .flatMap((accessToken) => {
          const oldHeaders = operation.getContext().headers;
          operation.setContext({
            headers: {
              ...oldHeaders,
              authorization: `Bearer ${accessToken}`,
            },
          });

          return forward(operation);
        });
    }
  }
});

const httpLink = new HttpLink({ uri: `${apiBaseUrl}/graphql/` });

const client = new ApolloClient({
  cache: new InMemoryCache({
    typePolicies: {
      Query: {
        fields: {
          reservationUnits: relayStylePagination(),
        },
      },
    },
  }),
  link: from([errorLink, authLink, httpLink]),
  ssrMode: typeof window === undefined,
  defaultOptions: {
    watchQuery: {
      errorPolicy: "ignore",
    },
    query: {
      errorPolicy: "ignore",
      fetchPolicy: typeof window === undefined ? "no-cache" : "cache-first",
    },
  },
});

export default client;
