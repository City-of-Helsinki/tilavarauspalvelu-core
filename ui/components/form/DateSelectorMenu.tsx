import { IconAngleLeft, IconAngleRight, IconCalendarPlus } from "hds-react";
import { camelCase, startCase } from "lodash";
import React, { ChangeEvent, MutableRefObject } from "react";
import { useTranslation } from "react-i18next";
import styled from "styled-components";
import { breakpoint } from "../../modules/style";
import Checkbox from "./Checkbox";
import DateRangePicker from "./DateRangePicker";

export const testIds = {
  menu: "date-selector-menu",
};

interface Props {
  backBtnRef?: MutableRefObject<HTMLButtonElement | null>;
  customDatesBtnRef?: MutableRefObject<HTMLButtonElement | null>;
  dateTypes: string[];
  dateTypeOptions: string[];
  endDate: Date | null;
  isCustomDate: boolean;
  isOpen: boolean;
  name: string;
  onChangeDateTypes: (value: string[]) => void;
  onChangeEndDate: (date: Date | null) => void;
  onChangeStartDate: (date: Date | null) => void;
  onCloseMenu: () => void;
  startDate: Date | null;
  toggleIsCustomDate: () => void;
}

const Wrapper = styled.div`
  background: var(--color-white);
  color: var(--color-black-90);
  box-shadow: 0px 0px 8px 0px rgba(0, 0, 0, 0.2);
  z-index: 1;
  min-width: fit-content;

  @media (min-width: ${breakpoint.m}) {
    position: absolute;
    top: 100%;
    left: 0;
    right: 0;
  }
`;

const CheckboxWrapper = styled.div`
  padding: 1rem;

  div {
    margin: 1rem 0;

    &:first-child {
      margin-top: 0;
    }

    &:last-child {
      margin-bottom: 0;
    }
  }
`;

const Btn = styled.button<{ $hidden: boolean }>`
  cursor: pointer;
  background: transparent;
  border: none;
  display: ${({ $hidden }) => ($hidden ? "none" : "flex")};
  width: 100%;
  padding: 1rem;
  align-items: center;

  svg:first-child {
    margin-right: 1rem;
  }

  svg:last-child {
    margin-left: 0.75rem;
  }

  &:focus {
    outline: 2px solid var(--color-black);
    outline-offset: -2px;
  }

  &.selectDates {
    border-top: 1px solid var(--color-silver);
    font-family: var(--font-medium);
    font-weight: 500;
  }

  &.back {
    font-family: var(--font-medium);
    font-weight: 500;
    border-bottom: 1px solid var(--color-silver);

    svg:first-child {
      margin-right: 0.75rem;
    }
  }

  &.close {
    border-top: 1px solid var(--color-silver);

    .buttonText {
      text-align: center;
    }
  }
`;

const BtnText = styled.div`
  flex: 1 1 0%;
  text-align: left;
`;

const CustomDateWrapper = styled.div`
  padding: 1rem;
`;

const DateSelectorMenu = ({
  backBtnRef,
  customDatesBtnRef,
  dateTypes,
  dateTypeOptions,
  endDate,
  isCustomDate,
  isOpen,
  name,
  onChangeDateTypes,
  onChangeEndDate,
  onChangeStartDate,
  onCloseMenu,
  startDate,
  toggleIsCustomDate,
}: Props): JSX.Element => {
  const { t } = useTranslation();

  const handleCheckboxChange = (event: ChangeEvent<HTMLInputElement>) => {
    if (dateTypes.indexOf(event.target.value) !== -1) {
      onChangeDateTypes(
        dateTypes.filter((item) => item !== event.target.value)
      );
    } else {
      onChangeDateTypes([...dateTypes, event.target.value]);
    }
  };

  if (!isOpen) return null;

  return (
    <Wrapper data-testid={testIds.menu}>
      {!isCustomDate && (
        <CheckboxWrapper>
          {dateTypeOptions.map((option) => {
            return (
              <Checkbox
                key={option}
                checked={dateTypes.indexOf(option) !== -1}
                id={`name_${option}`}
                label={t(
                  `dateSelector:dateType${startCase(camelCase(option)).replace(
                    / /g,
                    ""
                  )}`
                )}
                name={name}
                onChange={handleCheckboxChange}
                value={option}
              />
            );
          })}
        </CheckboxWrapper>
      )}

      <Btn
        ref={customDatesBtnRef}
        className="selectDates"
        onClick={toggleIsCustomDate}
        type="button"
        $hidden={isCustomDate}
      >
        <IconCalendarPlus aria-hidden />
        <BtnText>{t("dateSelector:menu.buttonCustom")}</BtnText>
        <IconAngleRight aria-hidden />
      </Btn>

      <Btn
        ref={backBtnRef}
        className="back"
        onClick={toggleIsCustomDate}
        type="button"
        $hidden={!isCustomDate}
      >
        <IconAngleLeft aria-hidden />
        <BtnText>{t("dateSelector:menu.buttonBack")}</BtnText>
      </Btn>

      {isCustomDate && (
        <CustomDateWrapper>
          <DateRangePicker
            endDate={endDate}
            onChangeEndDate={onChangeEndDate}
            onChangeStartDate={onChangeStartDate}
            startDate={startDate}
          />
        </CustomDateWrapper>
      )}
      <Btn
        className="close"
        onClick={onCloseMenu}
        type="button"
        $hidden={isCustomDate}
      >
        <BtnText>{t("dateSelector:menu.buttonClose")}</BtnText>
      </Btn>
    </Wrapper>
  );
};

export default DateSelectorMenu;
