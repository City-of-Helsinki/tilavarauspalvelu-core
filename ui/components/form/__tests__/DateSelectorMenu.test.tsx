import * as React from "react";

import { DATE_TYPES } from "../../../modules/const";
import { render } from "../../../test/testUtils";
import DateSelectorMenu from "../DateSelectorMenu";

jest.mock("next/config", () => () => ({
  serverRuntimeConfig: {},
  publicRuntimeConfig: {},
}));

jest.mock("next-i18next", () => ({
  useTranslation: () => {
    return {
      t: (str: string) => str,
    };
  },
}));

test("matches snapshot", () => {
  const { container } = render(
    <DateSelectorMenu
      dateTypes={["type1", "type2"]}
      dateTypeOptions={[
        DATE_TYPES.TODAY,
        DATE_TYPES.TOMORROW,
        DATE_TYPES.THIS_WEEK,
        DATE_TYPES.WEEKEND,
      ]}
      endDate={new Date("2019-09-31")}
      isCustomDate={false}
      isOpen
      name="date"
      onChangeDateTypes={jest.fn()}
      onChangeEndDate={jest.fn()}
      onChangeStartDate={jest.fn()}
      onCloseMenu={jest.fn()}
      startDate={new Date("2019-08-01")}
      toggleIsCustomDate={jest.fn()}
    />
  );

  expect(container.firstChild).toMatchSnapshot();
});
