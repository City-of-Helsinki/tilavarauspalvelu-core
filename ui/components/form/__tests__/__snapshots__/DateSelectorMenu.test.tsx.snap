// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="DateSelectorMenu__Wrapper-sc-4q8ipz-0 jXmzKr"
  data-testid="date-selector-menu"
>
  <div
    class="DateSelectorMenu__CheckboxWrapper-sc-4q8ipz-1 ilGTbW"
  >
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz Checkbox__StyledCheckbox-sc-14mc6iy-0 gYbgIB"
      style="--background-unselected: var(--color-white); display: block; line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_today"
        name="date"
        role="checkbox"
        type="checkbox"
        value="today"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_today"
      >
        dateSelector:dateTypeToday
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz Checkbox__StyledCheckbox-sc-14mc6iy-0 gYbgIB"
      style="--background-unselected: var(--color-white); display: block; line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_tomorrow"
        name="date"
        role="checkbox"
        type="checkbox"
        value="tomorrow"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_tomorrow"
      >
        dateSelector:dateTypeTomorrow
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz Checkbox__StyledCheckbox-sc-14mc6iy-0 gYbgIB"
      style="--background-unselected: var(--color-white); display: block; line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_this_week"
        name="date"
        role="checkbox"
        type="checkbox"
        value="this_week"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_this_week"
      >
        dateSelector:dateTypeThisWeek
      </label>
    </div>
    <div
      class="Checkbox-module_checkbox__3L0GR checkbox_hds-checkbox__9HMCz Checkbox__StyledCheckbox-sc-14mc6iy-0 gYbgIB"
      style="--background-unselected: var(--color-white); display: block; line-height: 24px;"
    >
      <input
        class="Checkbox-module_input__3VZvy checkbox_hds-checkbox__input__1w0pu"
        id="name_weekend"
        name="date"
        role="checkbox"
        type="checkbox"
        value="weekend"
      />
      <label
        class="Checkbox-module_label__L5AN1 checkbox_hds-checkbox__label__3HoD3"
        for="name_weekend"
      >
        dateSelector:dateTypeWeekend
      </label>
    </div>
  </div>
  <button
    class="DateSelectorMenu__Btn-sc-4q8ipz-2 gDyqJS selectDates"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          d="M0 0h24v24H0z"
        />
        <path
          d="M20 14v3h3v2h-3v3h-2v-3h-3v-2h3v-3h2zM17 2a1 1 0 011 1v1h4v8h-2v-1H4v8h9v2H2V4h4V3a1 1 0 112 0v1h8V3a1 1 0 011-1zm3 4H4v3h16V6z"
          fill="currentColor"
        />
      </g>
    </svg>
    <div
      class="DateSelectorMenu__BtnText-sc-4q8ipz-3 eNxEJL"
    >
      dateSelector:menu.buttonCustom
    </div>
    <svg
      aria-hidden="true"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <polygon
          points="0 24 0 0 24 0 24 24"
        />
        <polygon
          fill="currentColor"
          points="13.5 12 8.5 7 10 5.5 16.5 12 10 18.5 8.5 17"
        />
      </g>
    </svg>
  </button>
  <button
    class="DateSelectorMenu__Btn-sc-4q8ipz-2 inhOyv back"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="Icon-module_icon__1Jtzj icon_hds-icon__1YqNC Icon-module_s__2WGWe icon_hds-icon--size-s__2Lkik"
      role="img"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <polygon
          points="24 0 24 24 0 24 0 0"
        />
        <polygon
          fill="currentColor"
          points="10.5 12 15.5 17 14 18.5 7.5 12 14 5.5 15.5 7"
        />
      </g>
    </svg>
    <div
      class="DateSelectorMenu__BtnText-sc-4q8ipz-3 eNxEJL"
    >
      dateSelector:menu.buttonBack
    </div>
  </button>
  <button
    class="DateSelectorMenu__Btn-sc-4q8ipz-2 gDyqJS close"
    type="button"
  >
    <div
      class="DateSelectorMenu__BtnText-sc-4q8ipz-3 eNxEJL"
    >
      dateSelector:menu.buttonClose
    </div>
  </button>
</div>
`;
